PSPs/drafts/psp-34.md

```diff
@@ -0,0 +1,914 @@
+      }
+    }
+  ],
+  "docs": [
+    "Returns the attribute of `id` for the given `key`.",
+    "",
+    "If `id` is a collection id of the token, it returns attributes for collection."
+  ],
+  "mutates": false,
+  "label": "PSP34Metadata::get_attribute",
+  "payable": false,
+  "returnType": {
+    "displayName": [
+      "Option"
+    ],
+    "type": "Option<[u8]>"
+  },
+  "selector": "0xf19d48d1"
+}
+```
+
+The list of required attributes for NFT should be defined in a separate 
+proposal based on the scope of the usage.
+
+### Events
+
+#### Transfer
+When a contract creates (mints) new tokens, `from` will be `None`.
+When a contract deletes (burns) tokens, `to` will be `None`.
+```json
+{
+  "args": [
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "from",
+      "type": {
+        "displayName": [
+          "Option"
+        ],
+        "type": "Option<AccountId>"
+      }
+    },
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "to",
+      "type": {
+        "displayName": [
+          "Option"
+        ],
+        "type": "Option<AccountId>"
+      }
+    },
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "id",
+      "type": {
+        "displayName": [
+           "Id"
+        ],
+        "type": "Id"
+      }
+    }
+  ],
+  "docs": [
+    "Event emitted when a token transfer occurs."
+  ],
+  "label": "Transfer"
+}
+```
+
+### Approval
+```json
+{
+  "args": [
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "from",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "to",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "id",
+      "type": {
+        "displayName": [
+           "Id"
+        ],
+        "type": "Id"
+      }
+    }
+  ],
+  "docs": [
+    "Event emitted when a token approve occurs."
+  ],
+  "label": "Approval"
+}
+```
+
+### ApprovalForAll
+```json
+{
+  "args": [
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "owner",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "docs": [],
+      "indexed": true,
+      "label": "operator",
+      "type": {
+        "displayName": [
+          "AccountId"
+        ],
+        "type": "AccountId"
+      }
+    },
+    {
+      "docs": [],
+      "indexed": false,
+      "label": "approved",
+      "type": {
+        "displayName": [
+          "bool"
+        ],
+        "type": "bool"
+      }
+    }
+  ],
+  "docs": [
+    "Event emitted when an operator is enabled or disabled for an owner.",
+    "The operator can manage all NFTs of the owner."
+  ],
+  "label": "ApprovalForAll"
+}
+```
+
+### AttributeSet
+```json
```

Comment:
What attribute? We only have `get_attribute` above and no `set_attribute`.

Reply:
Because `get_attribute` is permissionless method when `set_attribute` is permissioned